addons/gamification/models/gamification_badge.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from odoo import api, fields, models, SUPERUSER_ID, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class GamificationBadgeUser(models.Model):
+    """User having received a badge"""
+
+    _name = 'gamification.badge.user'
+    _description = 'Gamification user badge'
+    _order = "create_date desc"
+    _rec_name = "badge_name"
+
+    user_id = fields.Many2one('res.users', string="User", required=True, ondelete="cascade")
+    sender_id = fields.Many2one('res.users', string="Sender", help="The user who has send the badge")
+    badge_id = fields.Many2one('gamification.badge', string='Badge', required=True, ondelete="cascade")
+    challenge_id = fields.Many2one('gamification.challenge', string='Challenge originating', help="If this badge was rewarded through a challenge")
+    comment = fields.Text()
+    badge_name = fields.Char(related='badge_id.name')
+
+    @api.model
+    def create(self, vals):
+        self.env['gamification.badge'].browse(vals.get('badge_id')).check_granting()
+        return super(GamificationBadgeUser, self).create(vals)
+
+    @api.multi
+    def _send_badge(self):
+        """Send a notification to a user for receiving a badge
+
+        Does not verify constrains on badge granting.
+        The users are added to the owner_ids (create badge_user if needed)
+        The stats counters are incremented
+        :param ids: list(int) of badge users that will receive the badge
+        """
+        res = True
+        for badge_user in self:
+            template = self.env.ref('gamification.email_template_badge_received').get_email_template(badge_user.id)
```

Comment:
declare this outside of for loop and assign variable 
badge_template  = self.env.ref('gamification.email_template_badge_received').